main.cpp

```diff
@@ -77,6 +80,14 @@ int main() {
     for (const auto animal : linktimeplugin::plugins<PluginBase>()) {
         std::cout << animal->name() << ": " << animal->sound() << '\n';
     }
-
+    std::cout << "again with iterators\n";
+    std::for_each(
+        linktimeplugin::RegistrarBase<PluginBase>::begin(),
+        linktimeplugin::RegistrarBase<PluginBase>::end(),
```

Comment:
Compared to the for loop above, this iterator interface is much more complex to use, and the type name needs to be repeated. It would be nice to have something as elegant as the range-for syntax that still didn't have to create the std::vector returned by the plugins() function.

Reply:
Agreed. Typing it was very painful, I wouldn't want anyone else needing to do it.